Mathlib/Dynamics/TopologicalEntropy/NetEntropy.lean

```diff
@@ -0,0 +1,408 @@
+/-
+Copyright (c) 2024 Damien Thomine. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damien Thomine, Pietro Monticone
+-/
+import Mathlib.Dynamics.TopologicalEntropy.CoverEntropy
+
+/-!
+# Topological entropy via nets
+We implement Bowen-Dinaburg's definitions of the topological entropy, via nets.
+
+All is stated in the vocabulary of uniform spaces. For compact spaces, the uniform structure
+is canonical, so the topological entropy depends only on the topological structure. This will give
+a clean proof that the topological entropy is a topological invariant of the dynamics.
+
+A notable choice is that we define the topological entropy of a subset `F` of the whole space.
+Usually, one defines the entropy of an invariant subset `F` as the entropy of the restriction of the
+transformation to `F`. We avoid the latter definition as it would involve frequent manipulation of
+subtypes. Our version directly gives a meaning to the topological entropy of a subsystem, and a
+single lemma (`subset_restriction_entropy` in `.Morphism`) will give the equivalence between
+both versions.
+
+Another choice is to give a meaning to the entropy of `∅` (it must be `-∞` to stay coherent) and to
+keep the possibility for the entropy to be infinite. Hence, the entropy takes values in the extended
+reals `[-∞, +∞]`. The consequence is that we use `ℕ∞`, `ℝ≥0∞` and `EReal` numbers.
+
+We relate in this file `CoverEntropy` and `NetEntropy`. This file is downstream of
+`Mathlib.Dynamics.TopologicalEntropy.CoverEntropy` since the submultiplicative argument there
+(specifically `IsDynCoverOf.iterate_le_pow`) is more natural for covers.
+
+## Main definitions
+- `IsDynNetOf`: property that dynamical balls centered on a subset `s` of `F` are disjoint.
+- `netMaxcard`: maximal cardinal of a dynamical net. Takes values in `ℕ∞`.
+- `netEntropyInfEnt`/`netEntropyEnt`: exponential growth of `netMaxcard`. The former is
+defined with a `liminf`, the later with a `limsup`. Take values in `EReal`.
+
+## Implementation notes
+As when using covers, there are two competing definitions `netEntropyInfEnt` and
+`netEntropyEnt` in this file: one uses a `liminf`, the other a `limsup`. When using covers, we chose
+the `limsup` definition as the default. Because of lemmas `coverEntropyInf_eq_iSup_netEntropyInfEnt`
+and `coverEntropy_eq_iSup_netEntropyEnt`, we make the same choice here. Theorems about the
+topological entropy of invariant subsets will be stated using only `coverEntropy`.
```

Comment:
Could you emphasize that you are *not* defining `netEntropy` and `netEntropyInf` (as the supremum over all entourages of ...) because it coincides with the cover version, so there's no point in having two names for the same object?

Reply:
Done.